services/profile/controller/controller.go

```diff
@@ -196,3 +199,57 @@ func GetValidFilteredProfiles(w http.ResponseWriter, r *http.Request) {
 
 	json.NewEncoder(w).Encode(filtered_profile_list)
 }
+
+/*
+	IsEventRedeemed checks the appropriate table to check whether the given event id has already been redeemed. If the event is not in the table, it add to the array.
+*/
+func IsEventRedeemed(w http.ResponseWriter, r *http.Request) {
+	id := r.Header.Get("HackIllinois-Identity")
+	event_id := mux.Vars(r)["id"]
+
+	redemption_status, err := service.EventRedemptionStatus(id, event_id)
+
+	if err != nil {
+		errors.WriteError(w, r, errors.DatabaseError(err.Error(), "Could not check if event was redeemed for id "+id+"."))
+		return
+	}
+	if !redemption_status {
+		errors.WriteError(w, r, errors.DatabaseError(err.Error(), "Event with id "+id+" was already redeemed."))
```

Comment:
Could you write a t3st or manually verify that this indeed returns an http error code? Just want to double check that we are not returning a 200 with an error message in the body, which I think we might be doing in other places in our api.